pages/dkp/kommander/2.2/dkp-upgrade/upgrade-konvoy/enterprise/index.md

```diff
@@ -0,0 +1,164 @@
+---
+layout: layout.pug
+navigationTitle: DKP Enterprise
+title: Upgrade Konvoy for DKP Enterprise
+excerpt: Upgrade your Konvoy environment within the DKP Enterprise license.
+beta: false
+enterprise: false
+menuWeight: 30
+---
+
+## Prerequisites
+
+* Follow the steps listed in the [DKP upgrade overview][dkpup]
+
+* Ensure that all platform applications in the management cluster have been upgraded to avoid compatibility issues with the current Kubernetes version. This is done automatically when [upgrading Kommander][upgradekomm], so ensure that Kommander has been upgraded prior to Konvoy.
+
+* For air-gapped: Download the required bundles either at our [support site][supportsite] or [via the CLI][airgapbundle]
+
+The following infrastructure environments are supported:
+
+* Amazon Web Services (AWS)
+
+* Microsoft Azure
+
+* Pre-provisioned environments
+
+## Overview
+
+The Konvoy upgrade for DKP Enterprise encompasses the following three steps in sequential order:
+
+1. Upgrade the Cluster API (CAPI) components
+1. Upgrade the core addons
+1. Upgrade the Kubernetes version
+
+Run all three steps on the management cluster (Kommander cluster) first. Then, run the second and third step on additional managed clusters (Konvoy clusters), one cluster at the time.
+
+For a full list of DKP Enterprise features, see [DKP Enterprise][dkpenterprise].
+
+<p class="message--note"><strong>NOTE:</strong> You must maintain your attached clusters manually. Review the documentation from your cloud provider for more information.
+
+## Upgrade the CAPI components
+
+New versions of DKP come pre-bundled with newer versions of CAPI, newer versions of infrastructure providers or new infrastructure providers. When using a new version of the DKP CLI, upgrade all of these components first.
+
+If you are running on more than one management cluster (Kubernetes cluster provisioned with DKP), you must upgrade the CAPI Components on each of these clusters. Ensure your `dkp` configuration references the management cluster where you want to run the upgrade by setting the KUBECONFIG environment variable [to the appropriate kubeconfig file’s location][kubeconfig].
+
+<p class="message--note"><strong>NOTE:</strong> An alternative to initializing the KUBECONFIG environment variable is to use the <code>--kubeconfig=cluster_name.conf</code> flag, ensuring that Kommander upgrades on the workload cluster.</p>
+
+1. Run the following upgrade command for the CAPI components.
+
+```bash
+dkp upgrade capi-components
+```
+
+The command should output something similar to the following:
+
+```bash
+✓ Upgrading CAPI components
+✓ Waiting for CAPI components to be upgraded
+✓ Initializing new CAPI components
+✓ Deleting Outdated Global ClusterResourceSets
+```
+
+1. Once finished, verify that the components were upgraded using the `kubectl get provider` command.
+
+```bash
+kubectl get provider --all-namespaces
```

Comment:
This will change, I will provide the output once we have it.

Reply:
We can merge as to not hold this up, then update the output later.

Reply:
If we merge this way let's at least update to 
```suggestion
kubectl get providers.clusterctl.cluster.x-k8s.io --all-namespaces
```

We have at least two other CRDs called providers:
```
$ k api-resources | grep -i provider
providers                                                                      clusterctl.cluster.x-k8s.io/v1alpha3                           true         Provider
providers                                                                      externaldata.gatekeeper.sh/v1alpha1                            false        Provider
providers                                                                      notification.toolkit.fluxcd.io/v1beta1                         true         Provider

```